crates/runtime/src/analyser.rs

```diff
@@ -0,0 +1,15 @@
+//! Adapted from <https://github.com/YarnSpinnerTool/YarnSpinner/blob/da39c7195107d8211f21c263e4084f773b84eaff/YarnSpinner/Analyser.cs
+
+pub(crate) use self::default_analysers::*;
+pub use self::{context::*, diagnosis::*};
+use std::fmt::Debug;
+use yarn_slinger_core::prelude::*;
+
+mod context;
+pub(crate) mod default_analysers;
+mod diagnosis;
+
+pub trait CompiledProgramAnalyser: Debug {
+    fn diagnose(&mut self, program: &Program);
+    fn gather_diagnoses(&self) -> Vec<Diagnosis>;
```

Comment:
Gather is quite an odd word, maybe `collect`?

Reply:
https://github.com/yarn-slinger/yarn-slinger/pull/107/commits/ac7efce6bb522d82fce4b54682b9397f44bd586f